cupy/_sorting/sort.py

```diff
@@ -127,7 +127,25 @@ def msort(a):
     return sort(a, axis=0)
 
 
-# TODO(okuta): Implement sort_complex
+def sort_complex(a):
+    """ Sort a complex array using the real part first,
+     then the imaginary part.
+
+    Args:
+        a(cupy.ndarray): Array to be sorted.
+
+    Returns:
+        cupy.ndarray: sorted complex array.
+
+    .. seealso:: :func:`numpy.sort_complex`
+
+    """
+    if a.dtype.char == 'e':
+        a = a.astype('D')
+    a.sort()
```

Comment:
This line may change the input array. If it is intended behavior, please check also the input array after `xp.sort_complex` is called by using `testing.numpy_cupy_array_list_equal`.

Reply:
@asi1024 can you please review ?